frontend/src/concepts/metrics/kserve/types.ts

```diff
@@ -0,0 +1,31 @@
+import { ConfigMapKind } from '~/k8sTypes';
+import { KserveMetricsGraphTypes } from '~/concepts/metrics/kserve/const';
+
+export type KserveMetricsConfigMapKind = ConfigMapKind & {
+  data: {
+    supported: 'true' | 'false';
+    metrics: string;
+  };
```

Comment:
(I already approved, but then realized what you were saying above @christianvogt)
Is this type incorrect then, if `data` could be undefined in a configmap? maybe this should be `data?: {` and then optional chaining should be used where it is consumed

Reply:
Hmmm it shouldn't be optional for the contract we have with the backend - that's why I'm using a validation check when receiving the configmap - if those properties aren't there the UI will show an error state

Reply:
This check ensures that we don't ever run into the issue of those props being undefined.

https://github.com/alexcreasy/odh-dashboard/blob/b0dde63bf4040f9dc75037d1f44c35c612096742/frontend/src/concepts/metrics/kserve/useKserveMetricsConfigMap.ts

Reply:
@mturley I'm assuming this type is correct for kserve metrics as it expects both of these to be defined if the configmap is valid for this usecase.

In the case of a generic configmap this is invalid because data is optional. But once we go through a type guard check and confirm the configmap to be well formed, then it's correct that these data types aren't optional.

Reply:
(my last comment was made before your last two comments loaded, disregard that 😄 )
sounds good.